combadge/src/main.rs

```diff
@@ -61,7 +67,8 @@ fn beam_path(transporter: &mut TcpStream, path: &Path, beam_id: u64) -> std::io:
     Ok(())
 }
 
-fn beam_file(transporter: &mut TcpStream, path: &Path) -> std::io::Result<()> {
+fn beam_file(transporter: &mut TcpStream, maybe_base_path: Option<&Path>, path: &Path) -> std::io::Result<()> {
```

Comment:
I understand that you called it `maybe_base_path` to represent the fact that it's an optional, but such names are less common in Rust. You can simply call it `base_path`.